darts/models/forecasting/forecasting_model.py

```diff
@@ -316,23 +316,25 @@ def _fit_wrapper(
         series: TimeSeries,
         past_covariates: Optional[TimeSeries],
         future_covariates: Optional[TimeSeries],
+        **kwargs,
     ):
-        self.fit(series)
+        supported_params = inspect.signature(self.fit).parameters
+        kwargs_ = {k: v for k, v in kwargs.items() if k in supported_params}
+        if self.supports_past_covariates:
+            kwargs_["past_covariates"] = past_covariates
+        if self.supports_future_covariates:
+            kwargs_["future_covariates"] = future_covariates
+        self.fit(series, **kwargs_)
 
     def _predict_wrapper(
         self,
         n: int,
-        series: TimeSeries,
-        past_covariates: Optional[TimeSeries],
-        future_covariates: Optional[TimeSeries],
         num_samples: int,
-        verbose: bool = False,
-        predict_likelihood_parameters: bool = False,
-    ) -> TimeSeries:
-        kwargs = dict()
-        if self.supports_likelihood_parameter_prediction:
-            kwargs["predict_likelihood_parameters"] = predict_likelihood_parameters
-        return self.predict(n, num_samples=num_samples, verbose=verbose, **kwargs)
+        **kwargs,
+    ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+        supported_params = inspect.signature(self.predict).parameters
+        kwargs_ = {k: v for k, v in kwargs.items() if k in supported_params}
+        return self.predict(n, num_samples=num_samples, **kwargs_)
```

Comment:
Some thoghts here:
- Should we check here for past/future covariates similar to _fit_wrapper?

  E.g. add past/future_covariates as input parameters to _predict_wrapper and then:
  ```
  if self.uses_past_covariates:
      kwargs_["past_covariates"] = past_covariates
  ```

- as far as I can tell we could let `num_samples` be part of **kwargs.

- To avoid breaking changes, we can pop the supports_likelihood_parameter_prediciton from the `kwargs_` if it's not supported (as done before)

Reply:
You're right, all the `TorchForecastingModel` use the same `predict()` method but don't all support the same covariates. I agree for `num_samples`.

We don't need to do it for `predict_likelihood_parameters` as the `predict()` methods already indicate if the model support it or not (and we check if they are indeed probabilistic in the `predict()` call, which end up calling `GlobalForecastingModel.predict()`).